src/Sulu/Bundle/MediaBundle/Collection/Manager/CollectionManager.php

```diff
@@ -582,7 +587,46 @@ public function delete($id)
         $collectionTitle = $collectionMeta ? $collectionMeta->getTitle() : null;
         $locale = $collectionMeta ? $collectionMeta->getLocale() : null;
 
+        if (!$forceRemoveChildren) {
+            $user = $this->getCurrentUser();
+
+            if (null !== $user) {
+                $totalChildResourcesWithoutPermissions = $this->collectionRepository->countUnauthorizedChildCollectionsOfRootCollection(
+                    $id,
+                    $user,
+                    $this->permissions[PermissionTypes::DELETE]
+                );
+
+                if ($totalChildResourcesWithoutPermissions) {
+                    $childCollectionResourcesWithoutPermissions = $this->collectionRepository->findUnauthorizedChildCollectionResourcesOfRootCollection(
+                        $id,
+                        $user,
+                        $this->permissions[PermissionTypes::DELETE],
+                        3
+                    );
+
+                    throw new InsufficientChildPermissionsException(
+                        $childCollectionResourcesWithoutPermissions,
+                        $totalChildResourcesWithoutPermissions,
+                        PermissionTypes::DELETE
+                    );
+                }
+            }
+
+            $childResources = $this->findAllChildResourcesByRootCollection($id);
+
+            if (!empty($childResources)) {
+                $totalChildResources = $this->countGroupedResources($childResources);
+
+                throw new DependantResourcesFoundException(
+                    $childResources,
+                    $totalChildResources
+                );
+            }
```

Comment:
If we do the above I would do the same for this lines `checkDependantResourcesForDelete()`